domain/src/main/java/org/oppia/domain/oppialogger/exceptions/OppiaUncaughtExceptionHandler.kt

```diff
@@ -11,6 +11,8 @@ class OppiaUncaughtExceptionHandler @Inject constructor(
   private val consoleLogger: ConsoleLogger
 ) : Thread.UncaughtExceptionHandler {
 
+  private lateinit var existingUncaughtExceptionHandler: Thread.UncaughtExceptionHandler
```

Comment:
This seems like a misuse of lateinit var. From what I understand, lateinit var is meant to only be used when we guarantee initialization, just after construction of the class. In this case, you want to use a sentinel value to represent presence on whether the handler was correctly initialized by the calling API. Nullability is probably a better fit for this:

```kotlin
private var chainedUncaughtExceptionHandler: Thread.UncaughtExceptionHandler? = null
```

Reply:
I think since we now are going to use the onCreate function of the `ApplicationCreationListener`, we can initialise this there and keep the `lateinit var` in its place. Am I correct on this ?